### DISCUSSION TITLE
Allow live preview for content versioning with relationals

### DISCUSSION THREAD

**Discussion by alexvdvalk** [directus/directus#20700] [Feature Requests]
### Summary

Currently when fetching a content version through the rest api, the response payload of M2As and relationals includes the 
`{
create:[],
update:[],
delete:[]
}`

object instead of the 'main' item structure. This makes it difficult to use the live preview functionality with content versioning

### Basic Example

The update would include a function to generate the item object as if the version had been merged in. This would be in the form of an API response which users can consume in their live preview pages.

### Motivation

This use case will improve user experience for content editors. it will provide them the ability to live preview an item before promoting the draft version to main.

### Detailed Design

The main bulk of work here is to pull in the item in its current form, and merge in the relational changes instead of passing back the [update,create, delete] payload.

### Must Have:
- The response must have the same output structure as if querying the main item.

### Should Have:
- ability to specify which fields should be returned.

### Won't Have:
- Any form of diffing in the response

### Drawbacks

Recursive relationships could cause issues. 
Relational items could change independently of the version.

### Alternatives

This code could be implemented in downstream systems

### Adoption Strategy

This is a net new function that should not impact existing functionality.

### Unresolved Questions

How much impact would this have for the greater community

**Comment by alexvdvalk** [comment]
I guess the big challenge here is to simulate a related record that possibly doesn't exist yet. If the user is creating a relational record in the branch, we would have to assume or guess that any other fields in that record might be. For example, the "user_created", or the "date_created" field. How can we know the date_created when the record doesn't even exist!?

Or perhaps when creating the branch,  we could simulate a 'create' and store all the fields in the branch instead of the subset that the user creates. This also poses a risk where flows / hooks might run on item create and validate certain fields or connect to third party systems.

It's also possible to go into further relational nesting inside a branch and if we're simulating the creation of a bunch of records, what impact would this have on performance?

Would it be possible to actually create these records and have a column to determine if its a "real" record or part of a branch.

**Comment by d3v1an7** [comment]
Adding a +1 to this! The users (especially editiors) in our Directus instance _rely_ on live preview -- not being able to see how content changes impact layout/flow is a show stopper for versioning.

**Comment by ZaitcevIlya** [comment]
Hi guys, it seems this still an issue or I miss something? I'm asking as this is quite important feature for our release flow. Maybe anyone has a solution/hacks to solve/avoid this?

**Comment by JamesW1** [comment]
This is now supported via the [changes](https://github.com/directus/directus/releases) made to support relational versioned data across v11.11.0 and v11.12.0 releases.
